### PR TITLE
Add build script and add libcurl dependency 

### PR DESCRIPTION
**Pull request by EvgeniyPatlan**

### CODE REVIEW

File: build-ps/percona-server-5.7_builder.sh [hunk 1]
```diff
@@ -0,0 +1,825 @@
...
+gpgcheck=0
+enabled=1
+EOL
+    fi
+    return
+}
+
+add_percona_apt_repo(){
+  if [ ! -f /etc/apt/sources.list.d/percona-dev.list ]; then
+    cat >/etc/apt/sources.list.d/percona-dev.list <<EOL
```
Comment (abychko) [inline build-ps/percona-server-5.7_builder.sh]: https://jenkins.percona.com/apt-repo/percona-dev.list.template?

Comment (abychko) [reply build-ps/percona-server-5.7_builder.sh]: or https://repo.percona.com/apt/percona-release_latest.generic_all.deb?

File: build-ps/percona-server-5.7_builder.sh [hunk 2]
```diff
@@ -0,0 +1,825 @@
...
+    return
+}
+
+install_deps() {
+    if [ $INSTALL = 0 ]
+    then
+        echo "Dependencies will not be installed"
+        return;
+    fi
+    if [ ! $( id -u ) -eq 0 ]
```
Comment (abychko) [inline build-ps/percona-server-5.7_builder.sh]: a bit inverted logic. look at if [ $( id -u ) -ne 0 ]

File: build-ps/percona-server-5.7_builder.sh [hunk 3]
```diff
@@ -0,0 +1,825 @@
...
+        echo "It is not possible to instal dependencies. Please run as root"
+        exit 1
+    fi
+    CURPLACE=$(pwd)
+    
+    if [ "x$OS" = "xrpm" ]; then
+        RHEL=$(rpm --eval %rhel)
+        ARCH=$(echo $(uname -m) | sed -e 's:i686:i386:g')
+        add_percona_yum_repo
+        yum -y install http://www.percona.com/downloads/percona-release/redhat/0.1-4/percona-release-0.1-4.noarch.rpm || true
```
Comment (abychko) [inline build-ps/percona-server-5.7_builder.sh]: https://repo.percona.com/yum/percona-release-latest.noarch.rpm

File: build-ps/percona-server-5.7_builder.sh [hunk 4]
```diff
@@ -0,0 +1,825 @@
...
+            exit 1
+        fi
+    fi
+    return
+}
+
+add_percona_yum_repo(){
+    if [ ! -f /etc/yum.repos.d/percona-dev.repo ]
+    then
+        cat >/etc/yum.repos.d/percona-dev.repo <<EOL
```
Comment (abychko) [inline build-ps/percona-server-5.7_builder.sh]: https://jenkins.percona.com/yum-repo/percona-dev.repo with keys in the same dir

File: build-ps/percona-server-5.7_builder.sh [hunk 5]
```diff
@@ -0,0 +1,825 @@
...
+        apt-get -y install lsb-release wget
+        export DEBIAN_FRONTEND="noninteractive"
+        export DIST="$(lsb_release -sc)"
+
+	    until sudo apt-get update; do
+    	    sleep 1
+            echo "waiting"
+        done
+        apt-get -y purge eatmydata || true
+        echo "deb http://jenkins.percona.com/apt-repo/ ${DIST} main" > percona-dev.list
```
Comment (abychko) [inline build-ps/percona-server-5.7_builder.sh]: one more time? you have add_percona_apt_repo()

File: build-ps/percona-server-5.7_builder.sh [hunk 6]
```diff
@@ -0,0 +1,825 @@
...
+        apt-get -y install lsb-release libmecab-dev libncurses5-dev libreadline-dev libpam-dev zlib1g-dev libcurl4-openssl-dev
+        apt-get -y install libldap2-dev libnuma-dev libjemalloc-dev libeatmydata libc6-dbg valgrind libjson-perl python-mysqldb libsasl2-dev
+
+        apt-get -y install libmecab2 mecab mecab-ipadic
+        apt-get -y install build-essential devscripts
+        apt-get -y install cmake autotools-dev autoconf automake build-essential devscripts debconf debhelper fakeroot 
+
+
+    fi
+    if [ ! -d /usr/local/percona-subunit2junitxml ]; then
```
Comment (abychko) [inline build-ps/percona-server-5.7_builder.sh]: not sure if we need subunit. is it for Jenkins param/trunk builds?

Comment (abychko) [reply build-ps/percona-server-5.7_builder.sh]: we use awk for long time already

File: build-ps/percona-server-5.7_builder.sh [hunk 7]
```diff
@@ -0,0 +1,825 @@
...
+            fi
+        fi
+    fi
+
+    if [ ${DEBIAN_VERSION} = "xenial" ]; then
+        sed -i 's/export CFLAGS=/export CFLAGS=-Wno-error=date-time /' debian/rules 
+        sed -i 's/export CXXFLAGS=/export CXXFLAGS=-Wno-error=date-time /' debian/rules
+    fi
+
+    if [ ${DEBIAN_VERSION} = "stretch" ]; then
```
Comment (abychko) [inline build-ps/percona-server-5.7_builder.sh:655]: custom cflags like -Wno-* to disable errors are just workarounds to build the code of uneven quality. release code should be ready and it shouldn't require any adjustments. Roel and Laurynas can say more or make a corrections

File: build-ps/percona-server-5.7_builder.sh (hunk 6 above)
Comment (EvgeniyPatlan) [reply build-ps/percona-server-5.7_builder.sh]: done

File: build-ps/percona-server-5.7_builder.sh (hunk 5 above)
Comment (EvgeniyPatlan) [reply build-ps/percona-server-5.7_builder.sh]: done

File: build-ps/percona-server-5.7_builder.sh (hunk 4 above)
Comment (EvgeniyPatlan) [reply build-ps/percona-server-5.7_builder.sh]: done

File: build-ps/percona-server-5.7_builder.sh (hunk 3 above)
Comment (EvgeniyPatlan) [reply build-ps/percona-server-5.7_builder.sh]: done

File: build-ps/percona-server-5.7_builder.sh (hunk 2 above)
Comment (EvgeniyPatlan) [reply build-ps/percona-server-5.7_builder.sh]: done

File: build-ps/percona-server-5.7_builder.sh (hunk 1 above)
Comment (EvgeniyPatlan) [reply build-ps/percona-server-5.7_builder.sh]: done

**Comment by EvgeniyPatlan** [comment]
@abychko all comments has been fixed

**Review by abychko — APPROVED** [review]
looks good to me. OK to merge if test OK

**Pull request merged**